superset/connectors/sqla/models.py

```diff
@@ -731,6 +731,9 @@ def get_sqla_query(  # sqla
         groupby_exprs_sans_timestamp: OrderedDict = OrderedDict()
 
         if groupby:
+            # dedup columns while preserving order
+            groupby = list(dict.fromkeys(groupby))
```

Comment:
I'm confused, if `groupby` is a dict how can it have duplicate keys?

Reply:
Oh, I see... it's a list, not a dict. The `fromkeys` threw me off, since I was expecting it to receive an argument that has keys.

Reply:
oh it's a way do dedup while preserving ordering in py3 (now that dict preserve order)